saml/saml.go

```diff
@@ -33,7 +33,7 @@ func Get(data string) (a ARN, err error) {
 
 	switch len(arns) {
 	case 0:
-		err = errors.New("no valid AWS roles were returned")
+		err = errors.New("no valid AWS roles were returned, check for extra characters on the IDP config")
```

Comment:
I wouldn't change the error text here. The `Get()` function has no idea why `extractArns()` returned zero entries. Extra characters is just one reason for an empty result and we can't foresee all of them.

In case we *did* want to provide more insights as to why we got zero entries, a better thing to do would be to have `extractArns()` return an error along with the result and inspect this in `Get()`. Then, a user would get a typical "chained" error at the top-most level, similar to the following (fake) error:

```
Could not get credentials: SAML error: no valid AWS roles were returned: invalid characters in IDP string
```

The above is only if `extractArns` can know for sure that the problem it had was due to invalid characters. If that's not the case it should return a more general error.

Reply:
yeah, I guess you're right.